WooCommerce/Classes/ViewRelated/Dashboard/Settings/Domains/DomainSelectorViewModel.swift

```diff
@@ -46,11 +46,19 @@ final class DomainSelectorViewModel: ObservableObject {
 
 private extension DomainSelectorViewModel {
     func observeDomainQuery() {
-        searchQuerySubscription = $searchTerm
+        let searchQuery = $searchTerm
             .map { $0.trimmingCharacters(in: .whitespacesAndNewlines) }
-            .debounce(for: .seconds(debounceDuration), scheduler: DispatchQueue.main)
             .filter { $0.isNotEmpty }
             .removeDuplicates()
+        // In order to always trigger a network request for the initial search query while supporting debounce
+        // for the following user input, the first search query is concatenated with the debounce version.
+        searchQuerySubscription = Publishers.Concatenate(
+            prefix: searchQuery.first(),
+            suffix: searchQuery
+                .debounce(for: .seconds(debounceDuration), scheduler: DispatchQueue.main)
```

Comment:
I'm not familiar with `Concatenate`, but should we add `dropFirst` for the suffix to avoid publishing again for the first element after the debounce?

Reply:
I thought so too and that's what I originally implemented with `dropFirst` to the debounce publisher, but then [a test case](https://buildkite.com/automattic/woocommerce-ios/builds/9542#0184cb60-1e1c-4712-b7d5-5b389b8b1ccc) failed where it asserts that a new search term right after the initial failure result still triggers a network request. That's why I put in another commit https://github.com/woocommerce/woocommerce-ios/pull/8271/commits/c0df7bbb1e242440c55945ffd6d3d047366be84a to fix this by dropping `dropFirst` and then adding `removeDuplicates()` to the concatenated publisher. I'll do a little more testing to see why the second query was missed in the dropFirst+debounce publisher.

Reply:
With a bit more testing, dropFirst+debounce by itself is emitting the expected values. So it looks like what causes the second value not to be emitted in the concatenated publisher is from the `Concatenate` behavior from the [official documentation](https://developer.apple.com/documentation/combine/publishers/concatenate?changes=_8):

> A publisher that emits all of one publisher’s elements before those from another publisher.

When the first value is emitted, `Publishers.Concatenate` probably only has the first publisher `searchQuery.first()` working. Then `searchQuery.first()` ends, and the following values are emitted to the second publisher with dropFirst+debounce. That's why we don't need the `dropFirst` operator on the second debounced publisher.

Reply:
Thank you for the explanation!